aws-storage-s3/src/test/java/com/amplifyframework/storage/s3/StorageComponentTest.java

```diff
@@ -144,11 +145,12 @@ public void testGenerateUrlGetsPresignedUrl() throws StorageException {
      * invokes the registered AWSS3StoragePlugin instance and returns a
      * {@link StorageDownloadFileResult} with correct file path.
      * @throws StorageException when an error is encountered while downloading
+     * @throws IOException when a temporary file cannot be created
      */
     @Test
-    public void testDownloadToFileGetsFile() throws StorageException {
+    public void testDownloadToFileGetsFile() throws StorageException, IOException {
         final String fromRemoteKey = RandomString.string();
-        final String toLocalPath = RandomString.string();
+        final File toLocalFile = File.createTempFile("download", "data");
```

Comment:
We have a utility in the code base called `RandomTempFile`, I think that could be a good replacement to these `File.createTemptFile(..., ...)` calls:
1. It can throw some kind of a `E extends RuntimeException`, which would prevent the need for us to snake this `IOException` through, and onto every `@Test`;
2. Instead of passing `"data"` every time, we could let the `RandomTempFile` populate some random string data.

Reply:
The second parameter is actually a suffix, or file extension, so it's just creating an empty text file called `{prefix}.{suffix}`. `RandomTempFile` sounds better for upload tests, however for download tests we'd still like an empty file.

Reply:
`RandomTempFile` also can throw an `IOException`, so instead I left the throws declaration on each `@Test` if it was the only one, if not I up-scoped to `Exception`. Let me know if you had something else in mind.